### ISSUE TITLE
"Clear Leo data" cursor does not signify it's clickable

### ISSUE THREAD

**Issue by stevelaskaridis** [brave/brave-browser#32529]
## Description 

Cursor on Leo's "Clear Leo data" does not suggest that the option is clickable.

## Steps to Reproduce

   1. Go to brave://settings/leo-assistant
   2. Hover over "Clear Leo data"
   3. Cursor is a caret instead of a pointer

## Actual result:

<img width="1645" alt="Screenshot 2023-08-25 at 17 57 25" src="https://github.com/brave/brave-browser/assets/5221013/7255c7b4-becf-4a2a-8006-29868131243b">

## Expected result:

<img width="1645" alt="Screenshot 2023-08-25 at 17 59 05" src="https://github.com/brave/brave-browser/assets/5221013/64d30211-369b-42de-800c-95ae49e9f4f6">

## Reproduces how often: 

Easily reproduced

## Desktop Brave version:

1.59.47 Chromium 116.0.5845.114 nightly (arm64), on OS X Ventura 13.4

## Android Device details:
   - Install type (ARM, x86): n/a
   - Device type (Phone, Tablet, Phablet): n/a
   - Android version: n/a

## Version/Channel Information:

- Can you reproduce this issue with the current release? no
- Can you reproduce this issue with the beta channel? no
- Can you reproduce this issue with the nightly channel? yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? no
- Does the issue resolve itself when disabling Brave Rewards? no
- Is the issue reproducible on the latest version of Chrome? no

**Comment by stephendonner** [comment]
@stevelaskaridis @petemill this needs a milestone

**Comment by stevelaskaridis** [comment]
I don't seem to be able to allocate a milestone to the issue.

**Comment by bsclifton** [comment]
@stephendonner just set the milestone now!

**Comment by Uni-verse** [comment]
@kjozwiak This feature is still in progress for Android and will not be included in the Leo MVP launch (1.61.x)

**Comment by kjozwiak** [comment]
> @kjozwiak This feature is still in progress for Android and will not be included in the Leo MVP launch (1.61.x)

Thanks, removing the `OS/Android` label as it sounds like the above isn't applicable for `Android` and won't even be included in `1.61.x` as per the above.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.62.15 Chromium: 119.0.6045.59 (Official Build) nightly (64-bit)
-- | --
Revision | 0d36c4485cf4ac1702afb55ae44c5b43ee85b30e
OS | Windows 11 Version 22H2 (Build 22621.2506)
```

Ensured that `Clear Leo data` via `brave://settings/leo-assistant` looks clickable as per the following:

**`Example`** | **`Example`**
---------------|-----------------
![Screenshot 2023-10-31 224544](https://github.com/brave/brave-browser/assets/2602313/91d63c80-38fb-4e79-a2bc-472ef318af06) | ![Screenshot 2023-10-31 224558](https://github.com/brave/brave-browser/assets/2602313/bc1de1c0-4d96-4f68-af95-96f9aa1a3459)
